torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -502,6 +502,40 @@ def _sfdp_replacement_18(query, key, value, causal_mask, dropout_p):
     )
 
 
+def _sfdp_pattern_19(query, key, value, causal_mask, attn_mask, dropout_p):
+    # for token-classification+gpt2 / text-generation+gpt2
+    attn_weights = torch.matmul(query, key.permute(0, 1, 3, 2))
+    inv_scale = torch.full(
+        [],
+        value.size(-1) ** 0.5,
+        dtype=attn_weights.dtype,
+        device=attn_weights.device,
+    )
+    attn_weights = attn_weights.div(inv_scale)
+    causal_mask_value = torch.full(
+        (), torch.finfo(query.dtype).min, dtype=query.dtype, device=query.device
```

Comment:
In the pattern, the causal mask dtype equals the query dtype..

Reply:
`causal_mask_value` uses the same type as query, but causal mask is bool.